src/kaleidoscope/hid.cpp

```diff
@@ -150,105 +176,132 @@ void sendKeyboardReport() {
   ConsumerControl.sendReport();
 }
 
-__attribute__((weak))
 void initializeConsumerControl() {
   ConsumerControl.begin();
 }
 
-__attribute__((weak))
 void pressConsumerControl(Key mappedKey) {
   ConsumerControl.press(CONSUMER(mappedKey));
 }
 
-__attribute__((weak))
 void releaseConsumerControl(Key mappedKey) {
   ConsumerControl.release(CONSUMER(mappedKey));
 }
 
 
-__attribute__((weak))
 void initializeSystemControl() {
   SystemControl.begin();
 }
 
-__attribute__((weak))
 void pressSystemControl(Key mappedKey) {
   SystemControl.press(mappedKey.keyCode);
 }
 
-__attribute__((weak))
 void releaseSystemControl(Key mappedKey) {
   SystemControl.release();
 }
 
 
 // Mouse events
 
-__attribute__((weak))
+#if !KALEIDOSCOPE_HIDADAPTOR_DISABLE_MOUSE
```

Comment:
I think I'd rather have these phrased as a positive rather than a double negative if it's not going to be a pain in the ass to make it overridable from the commandline. If it will be a pain in the ass, I'm willing to drop the request.

#if KALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE

Reply:
We can use a positive one, but then we need to change the default flags in `boards.txt`, and the override would be a `-UKALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE`. Not a pain in the ass to override, but slightly more complex.

Another option would be to override with `KALEIDOSCOPE_HIDADAPTOR_DISABLE_MOUSE`, but have something like this in a header:

```c++
#if KALEIDOSCOPE_HIDADAPTOR_DISABLE_MOUSE
#define KALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE 0
#else
#define KALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE 1
#endif

#if KALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE
// ....
#endif
```

So we'd still have enable-by-default, not just on the code level, but with the override conveying the same thing, but would remove the double-negative from the `#if`. I think that `-DKALEIDOSCOPE_HIDADAPTOR_DISABLE_MOUSE` makes more sense than `-UKALEIDOSCOPE_HIDADAPTOR_ENABLE_MOUSE`... undefining a flag that was added by `boards.txt` feels wrong.